src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/jclouds/openstack/OpenstackUtil.java

```diff
@@ -126,4 +128,23 @@ public void validateOpenstackInstanceParameters(Instance instance) {
         Validate.notNull(instance.getNumber(), "Instance parameter 'number' cannot be null");
         Validate.notBlank(instance.getNumber(), "Instance parameter 'number type' cannot be empty");
     }
+
+    /**
+     * @return The Operating system family either by default from jclouds or from the user metadata,
+     * created to patch the limitation caused by Jclouds for OpenStack (not taking the distro_family into account).
+     */
+    public static OsFamily getOpenStackOSFamily(Image image) {
+        if (image.getOperatingSystem().getFamily() == OsFamily.UNRECOGNIZED) {
+            if (image.getUserMetadata().containsKey("distro_family")) {
+                return OsFamily.fromValue(image.getUserMetadata().get("distro_family"));
+            } else {
+                log.warn("the image \"{}\" with ID \"{}\" is added with no recognized operating system family",
+                         image.getName(),
+                         image.getId());
+            }
+        } else {
+            return image.getOperatingSystem().getFamily();
+        }
+        return OsFamily.UNRECOGNIZED;
```

Comment:
It seems to me that L148 return can be directly moved after L141-143 (inside block)